src/app/store/modules/river-detail.js

```diff
@@ -0,0 +1,141 @@
+import actions from '@/app/store/actions'
+import mutations from '@/app/store/mutations'
+import { httpClient } from '@/app/global/services'
+import wkx from "wkx";
+
+export default {
+  namespaced: true,
+  state: {
+    error: false,
+    loading: false,
+    data: null,
+    refId: null
+  },
+  mutations: {
+    ...mutations,
+    ['GEOM_UPDATE_REQUEST'](state) {
+      Object.assign(state, { loading: true, error: null });
+    },
+
+    ['GEOM_UPDATE_SUCCESS'](state, payload) {
+      state.loading = false;
+      Object.assign(state.data, {
+        ...payload,
+      });
+    },
+
+    ['GEOM_UPDATE_ERROR'](state, payload) {
+      Object.assign(state, {
+        error: payload || true,
+        loading: false,
+      });
+    },
+  },
+  actions: {
+    ...actions,
+    async getProperty(context, id) {
+      try {
+        context.commit('DATA_REQUEST')
```

Comment:
So, no right or wrong here, and certainly one-offs and flag-flickers can live inline. The question was asked elsewhere about an opinion on where to put queries. For sharing with the other project I structure a light object-theme specific folder structure like this: 

- index.js (exported elements)
- models\
  - Reach.js
  - POI.js
  - GaugeOverview.js
- gql\
  - getReach.gql
  - updateReach.gql
- repository\
  - ReachRepo.js
- factory\
  - ReachFactory.js
- lib\ (optional)
  - ReachMapper.js

Model creation rules are in the Factory, superficial and misc operations on the model in lib, repository can save, restore, load, update, etc. and the model themselves are well defined, gql when heavy goes in the gql directory which allows reuse of fragments (e.g. update generally requests the same fields as get so that the object is properly reflected on update).

Reply:
Oh, and that structure is repeated per model-theme so 
\models\Users\
will have 
\models\Users\models\User.js
and \models\Users\repository\UserRepo.js

as well as 
\models\Reaches\ 
has
\models\Reaches\repository\ReachRepo.js
\models\Reaches\lib\ReachMapper.js 
etc.
The idea is to isolate the responsibility of the given models to reduce complexity.

Reply:
Tests are in the model-specific directory and are independently unit tested with the client repository mock. Operations are reproducible and libs try to not mutate the underlying objects.